### DISCUSSION TITLE
How does the plugin system work?  No idea how to load up ACE

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#171] [General]
Hi.

I understand that I need to git clone clog and clog-ace but I don't understand where to put it to be able to quickload :clog/tools.

I have tested out ql and ultralisp to automagically download and setup everything but loading up build returned an error and I decided that I would prefer to manually git clone clog and any amount of plugins so that I am a little more hands on and aware of the clog structure.

Can someone give me a hint on where plugins are supposed to be located in the clog folder?

Thanks.

**Comment by rabbibotton** [comment]
Using the plugins is simple. You just do (ql:quickload :clog-ace/tools) the key is the /tools on any plug in and it installs in builder. you use :clog-ace alone for anything run time, :clog-ace/tools for design time. That is it. You can load as many as you wish. Each is added to control pallet. You can also just add to your project. All the templates in builder now have /tools added for new projects. You just add depends-on to :my-proj for run-time and :my-proj/tools for design time.

The clog-builder does depend on clog-ace and automatically loads if for run-time use. You still have to (ql:quickload :clog-ace/tools) or add to your project's /tools for depends-on if want to be able to drag and drop an ace editor on to your web page or panel.

**Comment by aykaramba** [comment]
Ah!  Thank you kindly.

**Comment by rabbibotton** [comment]
https://docs.google.com/document/d/1nYyGVQwEZs5pq1f3GPkWFvwTvZyDp6fFfFTNyfLBdwY

That doc has instructions for using and creating
